Sources/AblyAssetTrackingInternal/Logging/DefaultHierarchicalLogHandler.swift

```diff
@@ -0,0 +1,30 @@
+import AblyAssetTrackingCore
+
+/// Provides an implementation of ``HierarchicalLogHandler`` by wrapping another instance of ``LogHandler``. It will insert a string such as "[asset-tracking.publisher.DefaultPublisher]" into the log messages, representing its list of subsystems.
+public struct DefaultHierarchicalLogHandler: HierarchicalLogHandler {
+    private var logHandler: LogHandler
+    private var subsystemNames: [String] // Lowest granularity first
+    
+    /// Creates an instance of ``DefaultHierarchicalLogHandler`` that writes messages to a given ``LogHandler``. The lowest-granularity subsystem of the returned log handler will always be "assetTracking".
+    /// - Parameters:
+    ///   - logHandler: A log handler to write messages to. If this is nil, the initializer will return nil.
+    ///   - subsystem: A subsystem to add to the list
+    public init?(logHandler: LogHandler?, subsystem: Subsystem? = nil) {
+        guard let logHandler = logHandler else {
+            return nil
+        }
+        self.logHandler = logHandler
+        self.subsystemNames = ["assetTracking"] + [subsystem?.name].compactMap { $0 }
```

Comment:
Nice one-liner approach to unwrapping the `subsystem?.name` optional value. I'm not sure if it's the most readable though. Wouldn't `if let name = subsystem?.name` be easier to read?

Up to you though, I don't really feel strongly either way.

Reply:
Fair point! I've changed it to your suggestion.